File: leetcode977/leetcode977.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int k = nums.size() - 1;
        vector<int> result(nums.size(), 0);
        for (int i = 0, j = k; i <= j;)
        {
            if (nums[j] * nums[j] > nums[i] * nums[i]){
                result[k] = nums[j] * nums[j];
                j--;
            }
            else{
                result[k] = nums[i] * nums[i];
                i++;
            }
            k--;
        }
        return result;
        
    } 
};

File: leetcode977/leetcode977.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        k = len(nums)
        res = [0] * k
        i, j = 0, k - 1
        for p in range(k - 1, -1, -1):
            x = nums[i] * nums[i]
            y = nums[j] * nums[j]
            if x > y:
                res[p] = x
                i += 1
            else:
                res[p] = y
                j -= 1
        return res